File: 백준/Silver/2628. 종이자르기/종이자르기.py
w, h = map(int, input().split())
n = int(input())
cut_w = [0, w]
cut_h = [0, h]

for _ in range(n):
    cr, num = map(int, input().split())
    
    if cr == 0:
        cut_h.append(num)
    else:
        cut_w.append(num)
        
cut_w.sort()
cut_h.sort()

max_w = 0
max_h = 0

for i in range(1, len(cut_w)):
    max_w = max(max_w, cut_w[i] - cut_w[i-1])
    
for i in range(1, len(cut_h)):
    max_h = max(max_h, cut_h[i] - cut_h[i-1])

print(max_w * max_h)

File: 백준/Silver/2628. 종이자르기/종이자르기.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(void)
{
    int w;
    int h;
    int n;
    cin >> w >> h;
    cin >> n;
    
    vector<int> cut_w;
    vector<int> cut_h;
    
    cut_w.push_back(0);
    cut_w.push_back(w);   
    
    cut_h.push_back(0);
    cut_h.push_back(h);
    
    for (int i = 0; i < n; i++) {
        int cr, num;
        cin >> cr >> num;
        
        if (cr == 0) {
            cut_h.push_back(num);    
        } else {
            cut_w.push_back(num);
        }
    }
    
    sort(cut_w.begin(), cut_w.end());
    sort(cut_h.begin(), cut_h.end());

    int max_w = 0;
    int max_h = 0;
    
    for (int i = 1; i < cut_w.size(); i++) {
        max_w = max(max_w, cut_w[i] - cut_w[i-1]);
    }

    for (int i = 1; i < cut_h.size(); i++) {
        max_h = max(max_h, cut_h[i] - cut_h[i-1]);
    }

    cout << max_w * max_h << endl;
    return 0;
}
